### PR TITLE
Fix nats reconnects

### PR DESCRIPTION
**Pull request by kobergj**
Repairs reconnects on the natsjs kv registry

New behaviour:
- when nats gets disconnected -> no change
- when nats gets reconnected -> registry reconnects correctly
- when nats connection is closed -> os.Exit as there is no way to recover without initializing everything

Fixes https://github.com/owncloud/ocis/issues/8783

### CODE REVIEW

File: ocis-pkg/natsjsregistry/registry.go [hunk 1]
```diff
@@ -150,6 +170,16 @@ func storeOptions(opts registry.Options) []store.Option {
 	natsOptions := nats.GetDefaultOptions()
 	natsOptions.Name = "nats-js-kv-registry"
 	natsOptions.User, natsOptions.Password = getAuth()
+	natsOptions.ReconnectedCB = func(_ *nats.Conn) {
+		if err := n.Init(); err != nil {
+			fmt.Println("cannot reconnect to nats")
+			os.Exit(1)
```
Comment (rhafer) [inline ocis-pkg/natsjsregistry/registry.go:176]: :boom: OUCH

I guess I understand why you're doing that. And I don't have a better solution at hand. But I still feel the need to say: This is ugly :smile: 

Would be nice if we could find a way for a more graceful error handling here.

**Review by rhafer — CHANGES_REQUESTED** [review]
Please change the changelog message wording. Other than that I am fine with it.

File: ocis-pkg/natsjsregistry/registry.go (hunk 1 above)
Comment (kobergj) [reply ocis-pkg/natsjsregistry/registry.go:176]: Yes I 100% agree. :heart: I'll fix it as soon as we have a proper solution.

**Review by rhafer — APPROVED** [review]
LGTM, but now Sonarcloud doesn't like you anymore ...

**Comment by kobergj** [comment]
No code changes - just force pushed to restart sonarcloud.

**Comment by wkloucek** [comment]
@micbar @kobergj is this suitable for a backport to oCIS 5.0.x ?

**Comment by rhafer** [comment]
@micbar Why do think it could be problematic regarding semver? It's a bugfix. It doesn't need new config. Neither does it break existing setups (does it?).

**Comment by micbar** [comment]
then let us backport.

**Pull request merged**